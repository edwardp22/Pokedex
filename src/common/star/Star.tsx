import React from "react";

import { Star as StarMUI } from "@mui/icons-material";

import styles from "./star.module.css";

export default function Star({
  selected,
  onClick,
}: {
  selected?: boolean;
  onClick: () => void;
}) {
  const { starSelected, starUnselected } = styles;

  return (
    <i onClick={() => onClick()}>
      <StarMUI className={selected ? starSelected : starUnselected} />
    </i>
  );
}
