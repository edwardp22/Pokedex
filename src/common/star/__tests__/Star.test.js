import renderer from "react-test-renderer";
import Star from "../Star";

it("should render selected star", () => {
  const component = renderer.create(<Star selected />);
  expect(component.toJSON()).toMatchSnapshot();
});

it("should render unselected star", () => {
  const component = renderer.create(<Star />);
  expect(component.toJSON()).toMatchSnapshot();
});
