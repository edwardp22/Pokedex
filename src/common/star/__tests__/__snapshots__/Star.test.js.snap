// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render selected star 1`] = `
<i
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium starSelected css-i4bv87-MuiSvgIcon-root"
    data-testid="StarIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
    />
  </svg>
</i>
`;

exports[`should render unselected star 1`] = `
<i
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium starUnselected css-i4bv87-MuiSvgIcon-root"
    data-testid="StarIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
    />
  </svg>
</i>
`;
